fix: OPTIC-1850: 'n is not a function' RUNTIME error occur on ls-develop when user tries to create a project

yyassi-heartex opened the pull request:
due to the upgrade to react 18 - we need to make sure that if a useEffect function only gets a destroy function when appropriate. we have use cases that shouldnt be returning to useEffect so we update it here

yyassi-heartex committed:
fix: OPTIC-1850: 'n is not a function' RUNTIME error occur on ls-develop when user tries to create a project

yyassi-heartex committed:
few more missed useEffects

nick-skriabin commented:
lgtm!

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14054261279)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1850/useeffect-return'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14054261279

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14054261279
